packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -292,6 +328,83 @@ const Autocomplete = <
       testId,
     ]
   );
+
+  const renderVirtualizedRow = ({
+    data,
+    index,
+    style,
+  }: ListChildComponentProps) => {
+    const baseOption = data[index];
+    /**
+     * react-window calculates the absolute positions of the list items, via an inline style, so
+     * we need to add it to each list item that is being rendered in the viewable list window.
+     * See here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     */
+    const optionItem = { ...baseOption, props: { ...baseOption.props, style } };
+    return optionItem;
+  };
+
+  const OuterListboxContext = createContext({});
+
+  const OuterListboxElementType = forwardRef<HTMLDivElement>((props, ref) => {
+    const outerProps = useContext(OuterListboxContext);
+    return <div ref={ref} {...props} {...outerProps} />;
+  });
+
+  function useResetCache(length: number) {
+    const ref = useRef<VariableSizeList>(null);
+    useEffect(() => {
+      if (ref.current) {
+        ref.current.resetAfterIndex(0, true);
+      }
+    }, [length]);
+    return ref;
+  }
+
+  const ListboxComponent = forwardRef<
+    HTMLDivElement,
+    HTMLAttributes<HTMLElement>
+  >(function (props, ref) {
+    const { children, ...other } = props;
+    const itemData: ReactElement[] = (children as ReactElement[]).flatMap(
+      (item: ReactElement & { children?: ReactElement[] }) =>
+        [item].concat(item.children || [])
+    );
+
+    // the height of an Odyssey autocomplete option item that is used to calculate height of window
+    const optionHeight = 45; //px
+
+    // The number of items (rows or columns) to render outside of the visible area for performance and scrolling reasons
+    const overscanRowCount = 8;
+
+    const gridRef = useResetCache(itemData.length);
+
+    return (
+      <ListboxContainer ref={ref}>
+        <OuterListboxContext.Provider value={other}>
+          <AutoSizer>
+            {({ height, width }) => (
+              <VariableSizeList
+                innerElementType="ul"
+                itemData={itemData}
+                itemCount={itemData.length}
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                itemSize={() => optionHeight}
```

Comment:
Sorry, I didn't notice this until now:
I'm wondering if this can be a constant function reference rather than one that gets created each render.

But even more than that, I think we should `useCallback` the entire `AutoSizer` render props function. That should account for any re-render situations.

Reply:
For sure! Let me make the change. Thanks for the suggestion.

Reply:
@KevinGhadyani-Okta I just pushed up a change for this - let me know if that's what you wanted done. Thanks!